packages/trpc/server/admin-router/update-organisation-member-role.ts

```diff
@@ -0,0 +1,218 @@
+import type { Prisma } from '@prisma/client';
+import { OrganisationGroupType, OrganisationMemberRole } from '@prisma/client';
+
+import { AppError, AppErrorCode } from '@documenso/lib/errors/app-error';
+import { generateDatabaseId } from '@documenso/lib/universal/id';
+import { getHighestOrganisationRoleInGroup } from '@documenso/lib/utils/organisations';
+import { prisma } from '@documenso/prisma';
+
+import { adminProcedure } from '../trpc';
+import {
+  ZUpdateOrganisationMemberRoleRequestSchema,
+  ZUpdateOrganisationMemberRoleResponseSchema,
+} from './update-organisation-member-role.types';
+
+/**
+ * Helper function to switch a member from one group to another atomically.
+ */
+const switchMemberGroup = async (
+  tx: Prisma.TransactionClient,
+  memberId: string,
+  fromGroupId: string,
+  toGroupId: string,
+) => {
+  await tx.organisationGroupMember.delete({
+    where: {
+      organisationMemberId_groupId: {
+        organisationMemberId: memberId,
+        groupId: fromGroupId,
+      },
+    },
+  });
+
+  await tx.organisationGroupMember.create({
+    data: {
+      id: generateDatabaseId('group_member'),
+      organisationMemberId: memberId,
+      groupId: toGroupId,
+    },
+  });
+};
```

Comment:
This would be the first instance of us taking a transaction as a parameter which feels kinda wrong, also the code is only used within this file so I'd probably just duplicate the code or consider how I can reorder my logic.

Reply:
fixed